dali/imgcodec/parsers/tiff.cc

```diff
@@ -13,18 +13,103 @@
 // limitations under the License.
 
 #include "dali/imgcodec/parsers/tiff.h"
+#include "dali/core/byte_io.h"
+#include "dali/imgcodec/util/exif.h"
 
 namespace dali {
 namespace imgcodec {
 
-ImageInfo TiffParser::GetInfo(ImageSource *encoded) const {
+constexpr int ENTRY_SIZE = 12;
+constexpr int WIDTH_TAG = 256;
+constexpr int HEIGHT_TAG = 257;
+constexpr int ORIENTATION_TAG = 274;
+constexpr int SAMPLESPERPIXEL_TAG = 277;
+constexpr int TYPE_WORD = 3;
+constexpr int TYPE_DWORD = 4;
+
+constexpr std::array<uint8_t, 4> le_header = {'I', 'I', 42, 0}, be_header = {'M', 'M', 0, 42};
```

Comment:
```suggestion
using tiff_magic_t = std::array<uint8_t, 4>;
constexpr tiff_magic_t le_header = {'I', 'I', 42, 0}, be_header = {'M', 'M', 0, 42};
```
you can reuse this typedef later, when reading.